devito/function.py

```diff
@@ -868,71 +868,45 @@ def coefficients(self):
 
         :returns: List of coefficients, eg. [b_11, b_12, b_21, b_22]
         """
-        # Grid indices corresponding to the corners of the cell
-        x1, y1, z1, x2, y2, z2 = sympy.symbols('x1, y1, z1, x2, y2, z2')
+        # Grid indices corresponding to the corners of the cell ie x1, y1, z1
+        indices1 = tuple(sympy.symbols('%s1' % d) for d in self.grid.dimensions)
+        indices2 = tuple(sympy.symbols('%s2' % d) for d in self.grid.dimensions)
+        # 1, x1, y1, z1, x1*y1, ...
+        indices = list(powerset(indices1))
+        indices[0] = (1,)
+        point_sym = list(powerset(self.point_symbols))
+        point_sym[0] = (1,)
+        # 1, px. py, pz, px*py, ...
+        A = []
+        ref_A = [np.prod(ind) for ind in indices]
+        # Create the matrix with the same increment order as the point increment
+        for i in self.point_increments:
```

Comment:
can you turn ``point_increments`` into a private variable, namely ``_point_increments``

Reply:
EDIT: ah, now I see, it's a property -- fine, it should be a private property, I think...